### PR TITLE
[Chef-16] Correct the rake task to pull in omnibus dependencies

### PR DESCRIPTION
**Pull request by neha-p6**
Signed-off-by: Neha Pansare <neha.pansare@progress.com>

## Description

Chef16 artifacts are now built with ruby31 using omnibus. 
This pr updates the rake task used to update dependencies for omnibus to support platform x64-mingw-ucrt for ruby31. 
and also updates the lock file for omnibus, now that ruby31 platform is added to pull in correct gems.

We started seeing windows build failures since [Bump omnibus from `5bc091a` to `d1fe3bb` in /omnibus by dependabot[bot] · Pull Request #13272 · chef/chef](https://github.com/chef/chef/pull/13272) was merged without running an adhoc build. The failure is:
```C:/opscode/omnibus-toolchain/embedded/lib/ruby/gems/3.1.0/gems/mixlib-shellout-3.2.7/lib/mixlib/shellout/windows.rb:21:in `require': cannot load such file -- win32/process (LoadError)```

(Ref build [here](https://buildkite.com/chef/chef-chef-chef-16-omnibus-adhoc/builds/60#018418ad-9049-49af-b29e-bcf77d477dfb))

This happens has the bump removed ucrt version of `mixlib-shellout` gem, it can no longer find it.
Turns out the ucrt supported gems were added manually https://github.com/chef/chef/commit/72e2d9d1a8c55ac3e51faec6cb045b885451d4cb, and since the rake task doesn't have the `x64-mingw-ucrt` platform listed, dependabot keeps removing them. 
This pr will make sure omnibus/Gemfile.lock is not updated manually.

Update1:  Now that ucrt platform is added, omnibus-software tries to look for ucrt gemspec for Chef, due to which windows builds fail when building chef gem.
We are fixing this here https://github.com/chef/omnibus-software/pull/1733

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

**Comment by neha-p6** [comment]
Adhoc build testing the change https://buildkite.com/chef/chef-chef-chef-16-omnibus-adhoc/builds/61

**Comment by neha-p6** [comment]
Adhoc build testing whether correct gemspec for windows is getting pulled or not: https://buildkite.com/chef/chef-chef-chef-16-omnibus-adhoc/builds/63

File: tasks/dependencies.rb [hunk 1]
```diff
@@ -32,8 +32,11 @@ def bundle_update_locked_multiplatform_task(task_name, dir)
     task task_name do
       Dir.chdir(dir) do
         Bundler.with_unbundled_env do
+          platform_list = "ruby x64-mingw32 x86-mingw32"
+          # chef-16 is now built using ruby31 with omnibus hence we need `x64-mingw-ucrt` platform ONLY for omnibus. The ruby version packaged inside chef continues to be ruby 2.7.5 (or as per omnibus_ovverrides.rb)
```
Comment (tpowell-progress) [inline tasks/dependencies.rb:36]: `omnibus_ovverrides.rb` -> `omnibus_overrides.rb`

File: cspell.json [hunk 2]
```diff
@@ -881,6 +881,7 @@
     "Overridable",
     "overridable",
     "Overriders",
+    "ovverrides",
```
[the remark this answers is not in the document]
Comment (neha-p6) [reply cspell.json]: Done

File: tasks/dependencies.rb (hunk 1 above)
Comment (neha-p6) [reply tasks/dependencies.rb]: Oops! How did I miss that. Thanks. Done.

**Comment by neha-p6** [comment]
Merging this since the corresponding omnibus-software pr https://github.com/chef/omnibus-software/pull/1733 is merged

**Pull request merged**